velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -931,6 +946,18 @@ void updateParquetWriterOptions(
               *hiveConfig->config(), core::QueryConfig::kSessionTimezone);
   }
 
+  parquetWriterOptions->parquetDataPageVersion =
```

Comment:
This would call getParquetDataPageVersion() twice if it can be acquired from the session property. I know you're copying from existing style, but it would be better to store the result from session property to a local variable, then check if it has value.

Reply:
@svm1 please address this comment.

Reply:
Thanks Ying, fair point - consolidated the redundant calls. Maybe the above blocks should also be rewritten (separately) to save the duplicate calls and help readability.